File: 数组/移除元素/26.删除有序数组中的重复项.py
#
# @lc app=leetcode.cn id=26 lang=python3
#
# [26] 删除有序数组中的重复项
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        n = len(nums)
        fast = 0
        slow = 0
        for fast in range(n):
            if nums[fast] != nums[slow]:
                slow += 1
                nums[slow] = nums[fast]
        return slow + 1

# @lc code=end

File: CProblems/数组/26.删除有序数组中的重复项.c
/*
 * @lc app=leetcode.cn id=26 lang=c
 *
 * [26] 删除有序数组中的重复项
 */

// @lc code=start
int removeDuplicates(int* nums, int numsSize) {
    // 双指针
    int slowIndex=0;
    for (int fastIndex=0; fastIndex<numsSize; fastIndex++) {
        if (nums[slowIndex] != nums[fastIndex]) {
            slowIndex++;
            nums[slowIndex] = nums[fastIndex];
        }
    }
    return slowIndex+1;
}
// @lc code=end

